File: Python/MiddleOfLinkedList.py
## Leetcode Problem - 876

class ListNode:
    def __init__(self,val=0,next=None) -> None:
        self.val = val
        self.next = next


class Solution:
    def middleNode(head):
        slow = head
        fast = head

        while fast.next and fast.next.next:
            slow = slow.next
            fast = fast.next.next

        if fast.next == None:
            return slow
        
        if fast.next.next == None:
            return slow.next
        

File: Javascript/MiddleOfLinkedList.js
// Leetcode Problem - 876

function ListNode(val,next){
    this.val = (val === undefined ? 0 : val)
    this.next = (next === undefined ? null : next)
}

var middleNode = function(head){
    let slow = head
    let fast = head

    while (fast.next && fast.next.next){
        slow = slow.next
        fast = fast.next.next
    }

    if (fast.next === null){
        return slow
    }else if (fast.next.next === null){
        return slow.next
    }
}